open_spiel/games/chess/chess_board.cc

```diff
@@ -451,88 +452,92 @@ Square ChessBoard<kBoardSize>::find(const Piece &piece) const {
   return InvalidSquare();
 }
 
-template <uint32_t kBoardSize>
-void ChessBoard<kBoardSize>::GenerateLegalMoves(
-    const MoveYieldFn &yield) const {
-  auto king_square = find(Piece{to_play_, PieceType::kKing});
+void ChessBoard::GenerateLegalMoves(
+    const MoveYieldFn &yield, Color color) const {
```

Comment:
Why the change of function signature to include color?

Reply:
In chess we only needed to generate legal moves for the color that actually can move. However, in darkchess the player can observe the chessboard after his own move and before the opponent's move. This observation is made from the squares he could move to (even though it's not his turn). So the dark chess observations are the reason why we need to generate "legal" moves for any player, not only the one that can actually move